clog/loggers.py

```diff
@@ -146,7 +142,19 @@ def __init__(self, host, port, retry_interval, report_status=None, logging_timeo
         self.retry_interval = retry_interval
         self.report_status = report_status or get_default_reporter()
         self.__lock = threading.RLock()
-        self._birth_pid = os.getpid()
+        self._pid = os.getpid()
+
+    def _reset_socket(self):
+        """Close and reconnect the socket, binding a new ephemeral port in the process"""
+        # The client/transport merely pass the close down to the TSocket class
+        # hence, closing the TSocket and forcing us to re-open will construct
+        # a new socket for communication to the Scribe server. Technically
+        # the old socket was still valid, but we would risk collision on writes
+        # with other processes writing to the socket. 
+        self.socket.close()
+        self.connected = False
+        self.last_connect_time = 0
+        self._maybe_reconnect()
```

Comment:
I don't believe so? I was debating leaving out the _maybe_reconnect() call from the reset function since we'll fall through to that "if not self.connected" check, but I felt it didn't fully encapsulate the idea of "resetting" the socket. But in this case if the _maybe_reconnect() fails, it should simply try to reconnect again once the function returns no?